homeassistant/components/sensor/wunderground.py

```diff
@@ -663,6 +663,8 @@ def __init__(self, rest, condition):
         self._entity_picture = None
         self._unit_of_measurement = self._cfg_expand("unit_of_measurement")
         self.rest.request_feature(SENSOR_TYPES[condition].feature)
+        self.entity_id = async_generate_entity_id(
```

Comment:
You are not allowed to call async functions from a sync context.

This is not executed in an async context (because it's `setup_platform` instead of `async_setup_platform`).

Reply:
Good catch! Fixed.